stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt

```diff
@@ -0,0 +1,828 @@
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.annotation.DrawableRes
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.getstream.sdk.chat.ChatUI
+import com.getstream.sdk.chat.DefaultBubbleHelper
+import com.getstream.sdk.chat.R
+import com.getstream.sdk.chat.adapter.AttachmentViewHolderFactory
+import com.getstream.sdk.chat.adapter.ListenerContainer
+import com.getstream.sdk.chat.adapter.MessageListItem
+import com.getstream.sdk.chat.adapter.MessageListItem.MessageItem
+import com.getstream.sdk.chat.adapter.MessageListItemAdapter
+import com.getstream.sdk.chat.adapter.MessageViewHolderFactory
+import com.getstream.sdk.chat.enums.GiphyAction
+import com.getstream.sdk.chat.navigation.destinations.AttachmentDestination
+import com.getstream.sdk.chat.utils.StartStopBuffer
+import com.getstream.sdk.chat.utils.extensions.inflater
+import com.getstream.sdk.chat.view.EndlessScrollListener
+import com.getstream.sdk.chat.view.IMessageListView
+import com.getstream.sdk.chat.view.MessageListView
+import com.getstream.sdk.chat.view.MessageListView.AttachmentClickListener
+import com.getstream.sdk.chat.view.MessageListView.GiphySendListener
+import com.getstream.sdk.chat.view.MessageListView.MessageClickListener
+import com.getstream.sdk.chat.view.MessageListView.MessageLongClickListenerView
+import com.getstream.sdk.chat.view.MessageListView.MessageRetryListener
+import com.getstream.sdk.chat.view.MessageListView.ReactionViewClickListener
+import com.getstream.sdk.chat.view.MessageListView.ReadStateClickListener
+import com.getstream.sdk.chat.view.MessageListView.UserClickListener
+import com.getstream.sdk.chat.view.MessageListViewStyle
+import com.getstream.sdk.chat.view.channels.ChannelsView
+import com.getstream.sdk.chat.view.dialog.MessageMoreActionDialog
+import com.getstream.sdk.chat.view.dialog.ReadUsersDialog
+import com.getstream.sdk.chat.view.messages.MessageListItemWrapper
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.ui.databinding.StreamMessageListViewBinding
+import io.getstream.chat.android.ui.messages.adapter.ListenerContainerImpl
+import kotlin.math.max
+import kotlin.math.min
+
+/**
+ * MessageListView renders a list of messages and extends the [RecyclerView]
+ * The most common customizations are
+ * - Disabling Reactions
+ * - Disabling Threads
+ * - Customizing the click and longCLick (via the adapter)
+ * - The list_item_message template to use (perhaps, multiple ones...?)
+ */
+public class MessageListView : ConstraintLayout, IMessageListView {
```

Comment:
Mostly a copy and past from the old MessageListView, with the change in the long click listener